File: gcc/others/selection_sort.c
#include<stdio.h>
#include<conio.h>

int main()
{
    int a[10], i, j, n, temp;

    printf("\n");
    printf("\n");
    printf("----- Selection Sort -----");
    printf("\n");
    printf("Enter the no. of elements in array: ");
    scanf("%d", &n);
    printf("Enter the elements\n");
    for(i=0;i<n;i++)
        scanf("%d", &a[i]);
    
    printf("\n");
    printf("\n");
    printf("--- OUTPUT ---\n");
    for(i=0;i<n;i++)
    {
        for(j=i+1;j<n;j++)
        {
            if(a[i]>a[j])
            {
                temp=a[i];
                a[i]=a[j];
                a[j]=temp;
            }
        }
    }

    printf("Sorted array:");
    for(i=0;i<n;i++)
    {
        printf("\t%d", a[i]);
    }
    printf("\n");
    printf("\n");
    getch();
}

File: java/others/lab/Learned1/selection_sort.java
import java.util.*;
import java.io.*;

public class selection_sort
{
    public static void main(String []args)
    {
        int i, j, temp, n;

        System.out.println("\n\n\SELECTION SORT\n");
        System.out.print("Enter the number of elements: ");
        Scanner sc=new Scanner(System.in);
        n=sc.nextInt();

        System.out.println("");
        System.out.println("Enter the elements: ");

        int A[] = new int[n];
        for(i=0;i<n;i++)
        {
            A[i]=sc.nextInt();
        }

        for(i=0;i<n;i++)
        {
            for(j=i+1;j<n;j++)
            {
                if(A[i]>A[j])
                {
                    temp=A[i];
                    A[i]=A[j];
                    A[j]=temp;
                }
            }
        }

        System.out.println("\n --- SORTED ARRAY ---");
        for(i=0;i<n;i++)
        {
            System.out.print("\t"+A[i]);
        }
        System.out.println("");

    }
}